qatrfm/environment.py

```diff
@@ -64,37 +64,31 @@ def __init__(self, image, tf_file=None, num_domains=1,
         shutil.copy(self.tf_file, self.workdir + '/env.tf')
         os.chdir(self.workdir)
         self.domains = []
-        self.networks = []
-        self.networks.append(self.get_network())
-        self.logger.debug("Using network {} for the domains".
-                          format(self.networks[0]))
-        # TODO: check how many networks the user needs and fill this array
-        #       accordingly
+        self.net_octet = self.get_network_octet()
+        self.logger.debug("Using network 10.{}.0.0/24".format(self.net_octet))
 
     @staticmethod
-    def get_network():
+    def get_network_octet():
         """
         Find a non-used network in the system
 
         To allow multiple environments co-exist, network ranges can't be
         hardcoded. Otherwise, new libvirt virtual networks can't be created.
```

Comment:
First I thought, it  doesn't make sense that an isolated network need to have a system-wide unique IP address. But I tried to create two isolated networks with virt-manager and I got 'Network already in use' so yes we need this workaround.

Reply:
Yes, and this system allows the environment to have up to 255 networks available for a single test. 10.X.Y.0/24 is a big range. For sure, we can do it in other ranges playing around with netmasks, but that's the simplest-working thing that came to my mind.